sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs

```diff
@@ -299,10 +300,15 @@ public void CreatesDefaultAzureCredential(
             {
                 configEntries.Add(new KeyValuePair<string, string>("managedIdentityResourceId", resourceIdValue));
             }
+            if (objectId)
+            {
+                configEntries.Add(new KeyValuePair<string, string>("managedIdentityObjectId", "objectId"));
+            }
+
             IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(configEntries).Build();
 
             // if both clientId and resourceId set, we expect an ArgumentException
-            if (clientId && resourceId)
+            if ((clientId && resourceId) || objectId)
```

Comment:
Would it be possible to update the comment, or move the check to a separate block?  Compound booleans are hard to mentally parse for humans to understand the code and know exactly what it's testing to make sure it's correct, or use a test failure to figure out what regressed in the code being validated by the test.

Reply:
updated the comment.